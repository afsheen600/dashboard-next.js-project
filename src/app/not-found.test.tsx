import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import NotFound from "./not-found";

describe("NotFound Page", () => {
  beforeEach(() => {
    render(<NotFound />);
  });

  it("renders the 404 heading", () => {
    // The Typography with variant="h2" renders as <h2>, not <h1>
    expect(
      screen.getByRole("heading", { name: "404", level: 2 })
    ).toBeInTheDocument();
  });

  it("renders the error message and emoji", () => {
    expect(
      screen.getByText(/sorry! something went wrong/i)
    ).toBeInTheDocument();
    expect(
      screen.getByText(/we could not find the page you were looking for/i)
    ).toBeInTheDocument();
    expect(screen.getByRole("img", { name: /sad/i })).toBeInTheDocument();
  });

  it("renders the Go back to Home button with correct link", () => {
    const homeLink = screen.getByRole("link", { name: /go back to home/i });
    expect(homeLink).toBeInTheDocument();
    expect(homeLink).toHaveAttribute("href", "/");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
  });
});
