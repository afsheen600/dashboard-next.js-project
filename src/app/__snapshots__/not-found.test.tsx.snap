// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NotFound Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-gmr9ql"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 css-u4oo4q-MuiPaper-root"
      style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-z510jb-MuiTypography-root"
      >
        404
      </h2>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-vso9nb-MuiTypography-root"
      >
        Sorry! Something went wrong 
        <span
          aria-label="sad"
          role="img"
        >
          ðŸ˜¢
        </span>
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1xz17g7-MuiTypography-root"
      >
        We could not find the page you were looking for.
      </p>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1los53i-MuiButtonBase-root-MuiButton-root"
        href="/"
        tabindex="0"
      >
        Go back to Home
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-gmr9ql"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 css-u4oo4q-MuiPaper-root"
      style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-z510jb-MuiTypography-root"
      >
        404
      </h2>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-vso9nb-MuiTypography-root"
      >
        Sorry! Something went wrong 
        <span
          aria-label="sad"
          role="img"
        >
          ðŸ˜¢
        </span>
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1xz17g7-MuiTypography-root"
      >
        We could not find the page you were looking for.
      </p>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1los53i-MuiButtonBase-root-MuiButton-root"
        href="/"
        tabindex="0"
      >
        Go back to Home
      </a>
    </div>
  </div>
</DocumentFragment>
`;
